common/executable_helpers/src/helpers.rs

```diff
@@ -84,9 +78,9 @@ pub fn setup_executable(
         is_logging_disabled,
         Some(config.base.node_async_log_chan_size),
     );
-
-    setup_metrics(&config.network.peer_id, &config);
-
+    for network in &config.networks {
+        setup_metrics(&network.peer_id, &config);
```

Comment:
Since the peer id of a node will be different on different networkconfigs, I don't think we have the bucketing problem you're mentioning. From the code, I don't there's any issue in calling `setup_metrics` multiple times with different peer ids.